### PR TITLE
Modifications to the mixbox_namespaces_revamp branch

### PR DESCRIPTION
**Pull request by bworrell**
@chisholm take a look at this and lemme know what you think.

### CODE REVIEW

File: mixbox/namespaces.py [hunk 1]
```diff
@@ -305,41 +298,34 @@ def set_preferred_prefix_for_namespace(self, ns_uri, prefix,
...
-        if ni.schema_location and incoming_schemaloc:
-            if ni.schema_location != incoming_schemaloc:
-                raise ConflictingSchemaLocationError(ni.uri,
-                                                     ni.schema_location,
-                                                     incoming_schemaloc)
-        elif ni.schema_location is None:
-                ni.schema_location = incoming_schemaloc or None
+        with theirs.  This modifies ni.
+        """
+        if ni.schema_location == incoming_schemaloc:  # TODO (bworrell): empty strings?
```
Comment (bworrell) [inline mixbox/namespaces.py]: I wasn't sure if we'd run into trouble if we compared an empty string to `None`. If so, we'd need to rethink the if/elif/else chain.

Comment (chisholm) [inline mixbox/namespaces.py]: If `ni.schema_location` is not None and `incoming_schemaloc` is None, then your code would raise an exception.  I don't think it should.  That means I already have a schemaloc and the incoming namespace doesn't have one.  It should just keep my existing schemaloc, i.e. do nothing, and not raise an exception.

This isn't an empty string <-> None comparison problem though.  There is also a difference of behavior here, where your code would treat empty string `ni.schema_location` as missing and update it from `incoming_schemaloc`, whereas mine would treat it as existing and not overwrite it.  Is that what you were getting at?

Comment (chisholm) [inline mixbox/namespaces.py]: Fyi, I just pushed an update to a test to reveal this bug, so you know what I'm talking about.  Mixbox's nosetests will fail on this branch until this is fixed.

Comment (bworrell) [inline mixbox/namespaces.py]: Should be fixed now!

**Pull request merged**